python/cuml/cuml/cluster/hdbscan/hdbscan.pyx

```diff
@@ -560,6 +561,17 @@ class HDBSCAN(UniversalBase, ClusterMixin, CMajorInputTagMixin):
         self.prediction_data_ptr = None
         self._cpu_to_gpu_interop_prepped = False
 
+    @property
+    def gen_min_span_tree(self):
+        return self._gen_min_span_tree
+
+    @gen_min_span_tree.setter
+    def gen_min_span_tree(self, value):
+        # If accelerator is active, we want to generate
+        # the mst to improve compatibility.
+        self._gen_min_span_tree = \
+            GlobalSettings().accelerator_active or value
```

Comment:
In general we should not set attributes to values that are different from the constructor argument.

I tried to work out why doing this increases compatibility, but couldn't the default of `gen_min_span_tree` in [HDBSCAN](https://hdbscan.readthedocs.io/en/latest/api.html) is also `False`. What was the thinking for making this default on when the accelerator is enabled?

Reply:
Over-zealousness of wanting the mst to be available always when I was testing locally, forgot that the original hdbscan package has the same gen_min_span_tree parameter, so I removed this logic :)